starlette/datastructures.py

```diff
@@ -252,26 +252,25 @@ def __init__(
         elif scope is not None:
             _items = parse_qsl(scope["query_string"].decode("latin-1"))
 
-        self._dict = {k: v for k, v in reversed(_items)}
+        self._dict = {k: v for k, v in _items}
```

Comment:
Note that `QueryParams([('a', 'first'), ('a', 'second')`])` will have different behavior here.

Reply:
Sorry, ignore this.